pennylane/resource/specs.py

```diff
@@ -12,19 +12,169 @@
+    batch, _ = qml.workflow.construct_batch(qnode, level=level)(*args, **kwargs)
+
+    # These values are the same for the whole batch, so we can just get them once
+    config = qml.workflow.construct_execution_config(qnode)(*args, **kwargs)
+    gradient_fn = config.gradient_method
+
+    for tape in batch:
+
+        info = specs_from_tape(tape, compute_depth)
+        info["num_device_wires"] = len(qnode.device.wires or tape.wires)
+        info["num_tape_wires"] = tape.num_wires
+        info["device_name"] = qnode.device.name
+        info["level"] = level
+        info["gradient_options"] = qnode.gradient_kwargs
+        info["interface"] = qnode.interface
+        info["diff_method"] = (
+            _get_absolute_import_path(qnode.diff_method)
+            if callable(qnode.diff_method)
+            else qnode.diff_method
+        )
+
+        if isinstance(gradient_fn, qml.transforms.core.TransformDispatcher):
+            info["gradient_fn"] = _get_absolute_import_path(gradient_fn)
+
+            try:
+                info["num_gradient_executions"] = len(gradient_fn(tape)[0])
+            except Exception as e:  # pylint: disable=broad-except
+                # In the case of a broad exception, we don't want the `qml.specs` transform
+                # to fail. Instead, we simply indicate that the number of gradient executions
+                # is not supported for the reason specified.
+                info["num_gradient_executions"] = f"NotSupported: {str(e)}"
+        else:
+            info["gradient_fn"] = gradient_fn
+
+        infos.append(info)
+
+    return infos[0] if len(infos) == 1 else infos
+
+
+# NOTE: Some information is missing from specs_qjit compared to specs_qnode
+def _specs_qjit(qjit, level, compute_depth, *args, **kwargs) -> SpecsDict:  # pragma: no cover
+    # pylint: disable=import-outside-toplevel
+    # Have to import locally to prevent circular imports as well as accounting for Catalyst not being installed
+    # Integration tests for this function are within the Catalyst frontend tests, it is not covered by unit tests
+
+    from catalyst.jit import QJIT
+
+    from ..devices import NullQubit
+
+    # TODO: Determine if its possible to have batched QJIT code / how to handle it
+
+    if not isinstance(qjit.original_function, qml.QNode):
+        raise ValueError("qml.specs can only be applied to a QNode or qjit'd QNode")
+
+    original_device = qjit.device
+    info = SpecsDict()
+
+    if level != "device":
+        raise NotImplementedError(f"Unsupported level argument '{level}' for QJIT'd code.")
+
+    # When running at the device level, execute on null.qubit directly with resource tracking,
+    # which will give resource usage information for after all compiler passes have completed
+
+    # TODO: Find a way to inherit all devices args from input
+    spoofed_dev = NullQubit(
+        target_device=original_device,
+        wires=original_device.wires,
+        shots=original_device.shots,
```

Comment:
Potentially need a way to silence the warning about passing shots to the device.  Though the qnode should already have the shots on it, so maybe we don't even need to specify the shots here.